### PR TITLE
extended GapToJsonStringForArangoDB

### PR DESCRIPTION
**Pull request by le-on**
so that it also splits input lines at commas if possible

### CODE REVIEW

File: gap/ArangoDB.gi [hunk 1]
```diff
@@ -165,13 +165,17 @@ InstallGlobalFunction( GapToJsonStringForArangoDB,
         c := c + Length( Positions( chunk, '\"' ) );
         
         if IsEvenInt( c ) then
-            Error( "splitting the input line for arangosh in the middle of a non-string is not supported yet\n" );
+            k := PositionNthOccurrence(Reversed(chunk), ',', 1);
+            if k = fail then
+                Error( "splitting the input line for arangosh in the middle of a non-string is not supported yet\n" );
```
Comment (mohamed-barakat) [inline gap/ArangoDB.gi]: Maybe one can then split the line at a white space. What do you think?

Comment (le-on) [reply gap/ArangoDB.gi]: That's a good idea! Or at a ':' if there is no whitespace to split at. I'll implement this tomorrow.

**Comment by le-on** [comment]
I just realized that the additional line splitting at commas or whitespace does not work properly: If there is a whitespace at the beginning of a chunk and a whitespace at the end of the next chunk, we would split at those, but the part in between could be longer than the allowed 7000 characters.
To resolve this, I rewrote the partitioning of the input line into chunks: Chunks now start at the last line break instead of just a multiple of 7000 in the input string.
What do you think of this solution?

**Comment by mohamed-barakat** [comment]
I very much like your PR, except for the modification of white spaces which your editor probably did automatically. Could you please push force a new commit only touching the lines you change?

**Comment by le-on** [comment]
Sorry about the whitespace changes. You were right, they were done automatically by my editor, I didn't even realize that.
I hope I did this right now, I'm not an expert in git :)

File: gap/ArangoDB.gi [hunk 2]
```diff
@@ -160,21 +160,41 @@ InstallGlobalFunction( GapToJsonStringForArangoDB,
...
-        
-        ## ASSUMPTION: only works if we are splitting a string
-        Append( str, Concatenation( chunk, "\\\n" ) );
         i := i + 1;
-        
+    
     until (i+1) * l > s;
     
     Append( str, string{[ (i * l + 1) .. s ]} );
+    until i + l > s;
```
Comment (mohamed-barakat) [inline gap/ArangoDB.gi:189]: There is some code duplication here, right?. In fact, GAP fails now load the package.

Comment (mohamed-barakat) [reply gap/ArangoDB.gi:189]: Could you please force push your fix?

**Comment by le-on** [comment]
Sorry about all the confusion. This should be it now.

**Comment by mohamed-barakat** [comment]
Excellent, thank you very much Luca.

**Pull request merged**